### ISSUE TITLE
Couldn't load images OpenCV(4.6.0)

### ISSUE THREAD

**Issue by kisst** [krupkat/xpano#112]
**Bug description**
```
~ ls -al *.jpeg  | wc -l 
3075
~ Xpano * --output=/tmp/panorama.jpeg
...
[2023-09-30 17:36:50.131] [XPano] [error] Couldn't load images OpenCV(4.6.0) ./modules/flann/src/miniflann.cpp:521: error: (-215:Assertion failed) (size_t)knn <= index_->size() in function 'runKnnSearch_'
```

All the images are part of the same one panorama picture, shoot with dwarf 2 .

**Desktop (please complete the following information):**
 - OS:  Linux
 - Version:  0.16.1
 - Method of installation: dpkg

**Comment by marcelloromani** [comment]
Have you tried with just a few images?

**Comment by kisst** [comment]
of course, with only a few images it works fine, but as you can expect based on the issue date, I did not held my breath until it's fixed, and moved on, but if it's helpful happy to re-test if there is a patch avaible

**Comment by marcelloromani** [comment]
Hmmm, perhaps it's a memory issue? Did you check the memory consumption of the process?
Also another test could be trying with half the images, if it fails half again, if not, upper half, etc. (binary search mode), to find out what's the max image count before it breaks.

**Comment by kisst** [comment]
If you mean run out of memory then I wouldn't expect as it existed rather quickly and I have 64GB so it would take some time before it fill it up all, either way I will create a new panorama and share that for testing, as I did not keep the original images just the render.

**Comment by krupkat** [comment]
Thank you @marcelloromani for the replies in this issue.

The actual error that comes from the miniflann module is something I have seen on much fewer images and I have a smaller reproducer now that I'd like to fix it there:

```
[2023-09-30 17:36:50.131] [XPano] [error] Couldn't load images OpenCV(4.6.0) ./modules/flann/src/miniflann.cpp:521: error: (-215:Assertion failed) (size_t)knn <= index_->size() in function 'runKnnSearch_'
```

Whether xpano can stitch ~ 3k images - I have my doubts, I have never tested such a scenario. Right now, the blending code requires all the images to be in memory at once (times 2x - 3x overhead) so this might be a blocker and would require a rewrite. There might also be issues with the optimization for camera poses for the 3k images, it is possible that it would have to be split into multiple smaller optimization steps.

Source images for such a large pano would be a very nice contribution :)

**Comment by kisst** [comment]
This weekend I will head up somewhere nice and take one then.  👍🏻

**Comment by kisst** [comment]
I hope you will get a good use of it, and let me know if you need me to add some specific license to a repo to be able to use it 
https://github.com/kisst/panorama-test-dataset
If you get to stitch it correctly do let me know, I am curious to see the end-result.
